site/content/en/docs/Tasks/namespaces_and_names.md

```diff
@@ -6,3 +6,245 @@ date: 2023-10-14
+metadata:
+  name: example
+  namespace: my-namespace
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config-m2mg5mb749
+  namespace: my-namespace
+data:
+  FOO: BAR
+```
+
+## Working with Names
+A prefix or suffix can be set for all Resources in a project with the `namePrefix` and `nameSuffix` fields. This sets a name prefix and suffix for both generated Resources (e.g. ConfigMaps and Secrets) and non-generated Resources.
+
+Resources such as Deployments and StatefulSets may reference other Resources such as ConfigMaps and Secrets in the Pod Spec. The name prefix and suffix will also propagate to Resource references in a project. Typical uses cases include Service reference from a StatefulSet, ConfigMap reference from a Pod Spec, and Secret reference from a Pod Spec.
+
+### Add Name Prefix
+The following example adds a prefix to the name of a Deployment and a generated ConfigMap.
+
+1. Create a Kustomization file.
+```yaml
+# kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namePrefix: foo-
+
+configMapGenerator:
+- name: my-config
+  literals:
+  - FOO=BAR
+
+resources:
+- deploy.yaml
+```
+
+2. Create a Deployment manifest.
+```yaml
+# deploy.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example
+```
+
+3. Add name prefix with `kustomize build`.
+```bash
+kustomize build .
+```
+
+The output shows that a prefix is added to the name of the Deployment and generated ConfigMap.
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: foo-my-config-m2mg5mb749
+data:
+  FOO: BAR
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: foo-example
+```
+
+### Add Name Suffix
+The following example adds a suffix to the name of a Deployment.
+
+1. Create a Kustomization file.
+```yaml
+# kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+nameSuffix: -bar
+resources:
+- deploy.yaml
+```
+
+2. Create a Deployment manifest.
+```yaml
+# deploy.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example
+```
+
+3. Add name suffix with `kustomize build`.
+```bash
+kustomize build .
+```
+
+The output shows that a suffix is added to the name.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example-bar
+```
+
+[`namespace`]: /docs/reference/api/kustomization-file/namespace/
+[`namePrefix`]: /docs/reference/api/kustomization-file/nameprefix/
+[`nameSuffix`]: /docs/reference/api/kustomization-file/namesuffix/
+
+### Propagate Name Prefix to Resource Reference
+The following example adds a prefix to the name of a Deployment and a generated ConfigMap. The generated ConfigMap name prefix is propagated to the Pod Spec.
+
+1. Create a Kustomization file.
+```yaml
+# kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namePrefix: foo-
+
+configMapGenerator:
+- name: special-config
+  literals:
+  - special.how=very
+
+resources:
+- deploy.yaml
+```
+
+2. Create a Deployment manifest.
```

Comment:
super nit: Can we indicate that the Deployment references the ConfigMap in this sentence?